MediaRanker complete

blip-lorist opened the pull request:
I'm very glad this was an exploratory project - I learned a TON from it! Stuck with the single model structure through the duration of the project, and I was able to remove excess controllers and shrink it down to just one controller. Very pleased with that. Thank you instructors for testing this app and trying to break it!

+1 for Bootstrap. I feel like this tool will save us SO much time with styling apps.

blip-lorist committed:
Updated .gitignore.

blip-lorist committed:
Rails app creation.

blip-lorist committed:
Created a medium model as an umbrella for movies, books, and albums.

blip-lorist committed:
Updated readme

blip-lorist committed:
Created a media controller

blip-lorist committed:
Created some seed data (slightly pokemon themed)

blip-lorist committed:
Forgot to add a 'media type' column to my db.

blip-lorist committed:
Added a few scopes.

blip-lorist committed:
Created a dynamic show for the media model.

blip-lorist committed:
Created a dynamic show for the media model.

blip-lorist committed:
WIP Creating a welcome layer for the main index page.

blip-lorist committed:
Decided to work without a welcome controller for now.

blip-lorist committed:
Consolidated all the dynamic show options under a single partial, media_details

blip-lorist committed:
Created a header partial.

blip-lorist committed:
Added header partial to views.

blip-lorist committed:
Created upvote feature.

blip-lorist committed:
Creating a table of movies on the movies index.

blip-lorist committed:
You can create a new medium record now.

blip-lorist committed:
Leveraged the params[:format] so that I could pull the new form out into a partial.

blip-lorist committed:
Pulled the media tables out into dynamic partials.

blip-lorist committed:
Created controllers for albums and books.

blip-lorist committed:
Dynamic new form and table partials now work for movies, books, and albums.

blip-lorist committed:
Added rspec to my gemfile.

blip-lorist committed:
Added rspec to my gemfile.

blip-lorist committed:
Created a spec for my media controller.

blip-lorist committed:
Added pry rails gem

blip-lorist committed:
Created test for creating new records.

blip-lorist committed:
Wrote test to ensure that the top scope ordered media correctly.

blip-lorist committed:
WIP edit action spec

blip-lorist committed:
Got edit working and views pulled out into partials.

blip-lorist committed:
Turns out I was on the right track with my rspec for the edit action. I just had a typo in my actual edit method. -_-

blip-lorist committed:
Omg got a delete test to work without too much hassle.

blip-lorist committed:
Fixed delete method redirect.

blip-lorist committed:
Added bootstrap rails to this project.

blip-lorist committed:
Changed up the rank and rank top ten scopes.

blip-lorist committed:
DRYed up my scopes.

blip-lorist committed:
Working on controller specs.

blip-lorist committed:
Created controller tests for GET #index

blip-lorist committed:
Created a test for upvote method. Thanks for your help @acmei

blip-lorist committed:
Removed excess code.

blip-lorist committed:
Created specs for model validations and scopes.

blip-lorist committed:
Starting on bootstrap styling with the site header.

blip-lorist committed:
WIP drying code up

blip-lorist committed:
Worked with @sallyamoore and @EliaMG on creating a partial for the main index using polymorphic paths.

blip-lorist committed:
Completed main index style.

blip-lorist committed:
Forgot a show controller spec, so I made one.

blip-lorist committed:
Did some random bootstrapy things to buttons.

blip-lorist committed:
Started a conversation with Jeremy about having one controller for this project.

blip-lorist committed:
Corrected my upvote method - was 'post', it's 'patch' now.

blip-lorist committed:
WIP one controller yay

blip-lorist committed:
YAAAY, got one controller, one model working!

blip-lorist committed:
Removed my extra controllers. Scary.

blip-lorist committed:
Edited the delete method redirect path.

blip-lorist committed:
Changed the buttons to links. Forgot you can add button styles to links.

blip-lorist committed:
Changed the buttons to links. Forgot you can add button styles to links.

blip-lorist committed:
Updated delete controller spec.

blip-lorist committed:
Changed buttons to links.

blip-lorist committed:
Cleaned up show routes.

blip-lorist committed:
Removed this partial.

blip-lorist committed:
I think I fixed the new form redirect else statement.

blip-lorist committed:
Style guide tweaks.

blip-lorist committed:
Pulled the top 10 number into its own local variable.

blip-lorist committed:
Last minute comments.

Hamled commented:
First off, I love how consolidated all the code is in this project! With the singular model class, simple controllers, and heavy use of partials this is a very lean (and thus more easily specified and understood) codebase.

blip-lorist committed:
Added header to the application, removed render partial tag from all views.

blip-lorist committed:
Added a custom new route.

blip-lorist commented:
Mahalo to @Hamled for the really helpful feedback! Got rid of the params[:format] thing and updated it to params[:media_type] - at the time of this project, I didn't understand what [:format] stored, so this info was really helpful in giving me some things to research.  

I've implemented all suggestions mentioned here in my latest pushed commits. Cheers!

Hamled commented:
Awesome! Glad to know the feedback has been helpful @lorainekv :)
